### PR TITLE
Move tests from `/tests` to `/src/tests` for `/chain`

### PR DESCRIPTION
**Pull request by pmnoxx**
Move tests from `/tests` to `/src/tests` for `/chain`,

As part of #5371

Once this gets merged. I'll work on next steps.

### CODE REVIEW

**Review by mzhangmzz — APPROVED** [review]

**Comment by mzhangmzz** [comment]
LTGM. Thanks!

**Comment by matklad** [comment]
Unless I am missing something, I'd expect this to also add

```rust
#[cfg(test)]
mod tests;
```

to `chain/network/src/lib.rs` and a `tests.rs` with corresponding `mod` declaration for each file. 

Right now it seems that these tests are no longer run at all? 

Coululd you double check that if, eg, you add an explicit `panic!` to a tests you can see the failure when running `cargo test`?

**Comment by pmnoxx** [comment]
@matklad Thanks, fixed.

**Comment by pmnoxx** [comment]
@matklad Can you take a look?

File: chain/client/Cargo.toml [hunk 1]
```diff
@@ -42,10 +42,10 @@ near-performance-metrics-macros = { path = "../../utils/near-performance-metrics
 near-vm-runner = { path = "../../runtime/near-vm-runner" }
 delay-detector = { path = "../../tools/delay_detector", optional = true }
 near-network-primitives = { path = "../network-primitives" }
+near-actix-test-utils = { path = "../../test-utils/actix-test-utils" }
+near-logger-utils = { path = "../../test-utils/logger" }
```
Comment (mzhangmzz) [inline chain/client/Cargo.toml:46]: Do we need to move these dependencies? I think we shouldn't because dev-dependencies should work for unit tests

Comment (pmnoxx) [reply chain/client/Cargo.toml]: @mzhangmzz Thanks, fixed.

**Review by matklad — APPROVED** [review]

**Comment by pmnoxx** [comment]
@bowenwang1996 @mm-near your approval is required.

**Review by bowenwang1996 — APPROVED** [review]

**Pull request merged**